pkg/urbit/noun/serial.c

```diff
@@ -70,12 +71,51 @@ _cs_jam_fib_mat(struct _cs_jam_fib* fib_u, u3_noun a)
     _cs_jam_fib_chop(fib_u, 1, 1);
   }
   else {
-    c3_w a_w = u3r_met(0, a);
-    c3_w b_w = c3_bits_word(a_w);
+    c3_w   a_w = u3r_met(0, a);
+    c3_w   b_w = c3_bits_word(a_w);
+    c3_w bit_w = fib_u->bit_w;
+
+    //  amortize overflow checks and reallocation
+    //
+    {
+      c3_w met_w = a_w + (2 * b_w);
+
+      if ( a_w > (UINT32_MAX - 64) ) {
+        u3m_bail(c3__fail);
+        return;
+      }
+
+      _cs_jam_fib_grow(fib_u, met_w);
+      fib_u->bit_w += met_w;
+    }
 
-    _cs_jam_fib_chop(fib_u, b_w+1, 1 << b_w);
-    _cs_jam_fib_chop(fib_u, b_w-1, a_w & ((1 << (b_w-1)) - 1));
-    _cs_jam_fib_chop(fib_u, a_w, a);
+    {
+      c3_w  src_w[2];
+      c3_w* buf_w = fib_u->sab_u->buf_w;
+
+      //  _cs_jam_fib_chop(fib_u, b_w+1, 1 << b_w);
+      //
+      {
+        c3_d dat_d = (c3_d)1 << b_w;
+        src_w[0]   = (c3_w)dat_d;
+        src_w[1]   = dat_d >> 32;
+
+        u3r_chop_words(0, 0, b_w + 1, bit_w, buf_w, 2, src_w);
```

Comment:
This would be more clearly expressed as
```c
c3_w src_w[] = {
  (c3_w)dat_d,
  dat_d >> 32,
};
u3r_chop_words(0, 0, b_w + 1, bit_w, buf_w, sizeof(src_w) / sizeof(*src_w), src_w);
```
without any loss in performance because `sizeof` is evaluated at compile time.

Reply:
Or, if you're opposed to inline use of `sizeof`, add a `c3_arr_len` macro to `c/defs.h`:
```c
#define c3_arr_len(arr) (sizeof(arr) / sizeof(*(arr)))
```

Reply:
It's impossible to argue that `sizeof(src_w) / sizeof(*src_w)` is clearer than `2`, when the length in question is the length of a "double word" in words.

Reply:
I disagree. That `src_w` has two elements and and that `2` is  an argument of `u3r_chop_words()` could be coincidental, and by using `sizeof()` in this way in place of `2` when calling `u3r_chop_words()`, you make that relationship explicit. 

I don't understand this opposition to what feels like plain common sense. There is no performance overhead, no added complexity, only a more explicitly communicated relationship between two different lines of code. I understand that magic numbers are used liberally throughout this code base, but using past behavior as a justification for present and future behavior is a tenuous argument at best.

Reply:
A consistent good style is better than a consistent flawed style is better than no consistent style.